### PR TITLE
Add S3 Credential datasource and resource

### LINKED ISSUE #285: Generate S3 credentials using Terraform
Hello,

I would like to be able to generate S3 credentials for a public cloud user from Terraform.

It works from OpenStack CLI or OVH manager.
But from Terraform it does not work using the OpenStack provider and the OVH provider does not provide a way to generate S3 credentials.

### Terraform Version
```
Terraform v1.2.6
on linux_amd64
+ provider registry.terraform.io/ovh/ovh v0.19.1
+ provider registry.terraform.io/terraform-provider-openstack/openstack v1.48.0
```

### Affected Resource(s)
- ovh_cloud_project_user
- openstack_identity_user_v3?
- openstack_identity_ec2_credential_v3?

### Terraform Configuration Files

```hcl
resource "ovh_cloud_project_user" "test" {
   service_name = var.service_name  # Public cloud project ID
   description = "Test"
}

# Get OpenStack ID
data "openstack_identity_user_v3" "test" {
  name = ovh_cloud_project_user.test.username
}

resource "openstack_identity_ec2_credential_v3" "test" {
  user_id = data.openstack_identity_user_v3.test.unique_id
}
```

The user used by the OpenStack provider has Administrator role.

### Expected Behavior
OVH provider should provide a way to generate S3 credentials.
Or openstack_identity_user_v3 and openstack_identity_ec2_credential_v3 should succeed.

### Actual Behavior
I got a permission issue with openstack_identity_user_v3

```sh
Error: Unable to query openstack_identity_user_v3: Request forbidden: [GET https://auth.cloud.ovh.net/v3/users?enabled=true&name=user-XXXX], error message: {"error": {"message": "You are not authorized to perform the requested action: identity:list_users.", "code": 403, "title": "Forbidden"}}
```

### Steps to Reproduce
1. `terraform apply`

Has anyone been able to generate S3 credentials from Terraform?

Thanks

### PR DESCRIPTION
**Pull request by yvigara**
Add the datasources and resource to create and retrieve S3 Credentials (AccessKeyId and SecretAccessKey) required to access S3 buckets. Fixes #285

### CODE REVIEW

File: website/docs/r/cloud_project_user_s3_credential.html.markdown [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# ovh_cloud_project_user_s3_credential
+
+Creates an S3 Credential for a user in a public cloud project.
+
+## Example Usage
+
+```hcl
+resource "ovh_cloud_project_user" "user" {
+ service_name = "XXX
+ description  = "my user for acceptance tests"
```
Comment (scraly) [inline website/docs/r/cloud_project_user_s3_credential.html.markdown:18]: In order to help user about how they can create this resource, adding a role_name can be useful.

For example:
```
role_name = "administrator"
```

or specify a list of role_names for the example

File: website/docs/r/cloud_project_user_s3_credential.html.markdown [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ service_name = ovh_cloud_project_user.user.service_name
+ user_id      = ovh_cloud_project_user.user.id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+- `service_name` - (Required) The id of the public cloud project. If omitted,
```
Comment (scraly) [inline website/docs/r/cloud_project_user_s3_credential.html.markdown:31]:
```suggestion
- `service_name` - (Required) The ID of the public cloud project. If omitted,
```

File: website/docs/r/cloud_project_user_s3_credential.html.markdown [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+- `user_id` - (Required) The id of a public cloud project's user.
```
Comment (scraly) [inline website/docs/r/cloud_project_user_s3_credential.html.markdown:34]:
```suggestion
- `user_id` - (Required) The ID of a public cloud project's user.
```

File: website/docs/d/cloud_project_user_s3_credentials.html.markdown [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+Use this data source to retrieve the list of all the S3 access_key_id associated with a public cloud project's user.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_user_s3_credentials" "user_1234_s3_credentials" {
+   service_name = "XXXXXX"
+   user_id      = "1234"
+}
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credentials.html.markdown:19]:
```suggestion
}

output "access_key_ids" {
    value = data.ovh_cloud_project_user_s3_credentials.myuser.access_key_ids
}
```

File: website/docs/d/cloud_project_user_s3_credentials.html.markdown [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+# ovh_cloud_project_user_s3_credentials (Data Source)
+
+Use this data source to retrieve the list of all the S3 access_key_id associated with a public cloud project's user.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_user_s3_credentials" "user_1234_s3_credentials" {
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credentials.html.markdown:16]:
```suggestion
data "ovh_cloud_project_user_s3_credentials" "myuser" {
```

File: website/docs/d/cloud_project_user_s3_credentials.html.markdown [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+   service_name = "XXXXXX"
+   user_id      = "1234"
+}
+```
+
+## Argument Reference
+
+- `service_name` - (Required) The id of the public cloud project. If omitted,
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credentials.html.markdown:24]:
```suggestion
- `service_name` - (Required) The ID of the public cloud project. If omitted,
```

File: website/docs/d/cloud_project_user_s3_credentials.html.markdown [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+- `user_id` - (Required) The id of a public cloud project's user.
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credentials.html.markdown:27]:
```suggestion
- `user_id` - (Required) The ID of a public cloud project's user.
```

File: website/docs/d/cloud_project_user_s3_credential.html.markdown [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+---
+
+# ovh_cloud_project_user_s3_credential (Data Source)
+
+Use this data source to retrieve the Secret Access Key of an Access Key ID associated with a public cloud project's user.
+
+## Example Usage
+
+```hcl
+data "ovh_cloud_project_user_s3_credentials" "s3_credentials" {
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credential.html.markdown:16]: Useless if we set manually the user_id in the `ovh_cloud_project_user_s3_credential` datasource.

File: website/docs/d/cloud_project_user_s3_credential.html.markdown [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+   service_name = "XXXXXX"
+   user_id      = "1234"
+}
+
+data "ovh_cloud_project_user_s3_credential" "s3_cred_1" {
```
Comment (scraly) [inline website/docs/d/cloud_project_user_s3_credential.html.markdown:21]: suggestion:

```
data "ovh_cloud_project_user_s3_credential" "mycred" {
   service_name = var.service_name
   user_id      = "244435"
   access_key_id = "xxxx"
}

output "myaccesskey" {
    value = ovh_cloud_project_user_s3_credential.s3_credential.access_key_id
}

output "mysecretaccesskey" {
    value = ovh_cloud_project_user_s3_credential.s3_credential.secret_access_key
    sensitive = true
}
```

or, if we want to use the datasource that list the access_key_id:

```
data "ovh_cloud_project_user_s3_credential" "mycred" {
   service_name = var.service_name
   user_id      = data.ovh_cloud_project_user_s3_credentials.myadmin.user_id
   access_key_id = data.ovh_cloud_project_user_s3_credentials.myadmin.access_key_ids[0]
}

output "myaccesskey" {
    value = ovh_cloud_project_user_s3_credential.s3_credential.access_key_id
}

output "mysecretaccesskey" {
    value = ovh_cloud_project_user_s3_credential.s3_credential.secret_access_key
    sensitive = true
}
```

**Comment by scraly** [comment]
Hi,
thanks for the PR, it's a good addition :)

I have an issue with the first acceptance test:
image

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by scraly** [comment]
About acceptance test, is it possible them like this example?
https://github.com/ovh/terraform-provider-ovh/blob/master/ovh/data_cloud_project_database_ip_restrictions_test.go

Thanks

**Comment by yvigara** [comment]
@scraly thanks a lot for the feedback. All the requested changes should be resolved now.

**Comment by yvigara** [comment]
Just to be sure, do you mean renaming the config variable to match the test function, keeping it as a simple string and moving the string formatting in the test function?

**Comment by scraly** [comment]
I mean moving the hash function to another file, next to the resource file.
Thanks to that in this new helpers file, in the future we can add another helper functions to that resource.

**Comment by scraly** [comment]
Sorry, I reviewed two PRs at the same time so my brain have switched :-D.
About the acceptance test I mean, yes, the const as string and how to pass the config.
Even if a lot of different people contribute to the repo, the goal is to try to keep a coherence when it is possible :).

**Comment by yvigara** [comment]
Should be all fixed now

**Review by scraly — APPROVED** [review]
Perfect, thanks :-) 
Acceptance tests are well written now and all the tests are in success.

**Pull request merged**